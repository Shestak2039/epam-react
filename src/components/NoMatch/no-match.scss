.no-match {
  display: flex;
  justify-content: center;
  flex-direction: column; 
  height: 100vh;
  background-color: white;
  align-items: center;
}

.no-match-link {
  text-decoration: none;
}
