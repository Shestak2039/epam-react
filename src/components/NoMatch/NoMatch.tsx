import React from 'react';
import { useLocation } from 'react-router';
import { Link } from 'react-router-dom';

import './no-match.scss';
import Button from '../common/Button/Button';

const NoMatch: React.FunctionComponent = () => {
    const location = useLocation();

    return (
        <div className="no-match">
            <h2>
                No match for <code>{location.pathname}</code>
            </h2>
            <Link to="/" className="no-match-link">
                <Button title={'GO BACK TO HOME'} primary={true} />
            </Link>
        </div>
    );
};

export default NoMatch;
