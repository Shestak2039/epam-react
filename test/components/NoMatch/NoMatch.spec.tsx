import React from 'react';
import { render } from '@testing-library/react';
import NoMatch from '../../../src/components/NoMatch/NoMatch';
import { BrowserRouter } from 'react-router-dom';


describe('NoMatch', () => {
    test('renders NoMatch snapshot', () => {
        const { asFragment } = render(<BrowserRouter><NoMatch /></BrowserRouter>);

        expect(asFragment()).toMatchSnapshot();
    });
});
