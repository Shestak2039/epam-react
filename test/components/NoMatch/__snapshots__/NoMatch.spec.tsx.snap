// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch renders NoMatch snapshot 1`] = `
<DocumentFragment>
  <div
    class="no-match"
  >
    <h2>
      No match for 
      <code>
        /
      </code>
    </h2>
    <a
      class="no-match-link"
      href="/"
    >
      <button
        class="button primary"
        data-testid="GO BACK TO HOME"
      >
        GO BACK TO HOME
      </button>
    </a>
  </div>
</DocumentFragment>
`;
